### DISCUSSION TITLE
Validation but do not restrict

### DISCUSSION THREAD

**Discussion by pimjansen** [edmundhung/conform#682] [General]
Im using conform together with zod in my remix application. All goes fine for the normal forms. However i just got a new requirement and have not a direct answer how to handle this.

The requirement is that a form should be able to be stored in an invalid state. The only thing i want to keep is actual validation so that it is still visible for the end user. So in general, validate the form client side (onInput) but just use a regular submit of the form

Anyone an idea how to handle this?
```js
    onValidate({ formData }) {
      return parseWithZod(formData, { schema });
    },
```

The code above matches the schema fine but there is no way to submit it regardless the outcome. Any ideas?

**Comment by Pouet--** [comment]
onValidate is call first at form submission, and the result of this function decides if the form is submitted or not.
The parseWithZod function returns a Submission object, and the "status" property has to be set to "success" to actually submit the form. Otherwise, it stops.

You can decide not to return the result of parseWithZod, but store it instead, and return the parseWithZod object with the status propery overrided and set to success :

A working example :
https://stackblitz.com/edit/remix-run-remix-xgkpsd?file=app%2Froutes%2F_index.tsx
